### PR TITLE
[SPARK-11019][streaming][flume] Gracefully shutdown Flume receiver th…

### PR DESCRIPTION
**Pull request by harishreedharan**
…reads.

Wait for a minute for the receiver threads to shutdown before interrupting them.

### CODE REVIEW

**Comment by tdas** [comment]
Why is this needed?

**Comment by harishreedharan** [comment]
This basically makes testing a bit more predictable. Sometimes, we end up hitting a situation where the last transaction is still not completed and the remaining data just stays stuck in Flume.

Also, it removes some unneeded `InterruptedException` from the logs

**Comment by harishreedharan** [comment]
Jenkins, test this please

**Comment by tdas** [comment]
Gotcha. How long generally does it take to shutdown? I mean, how long typically will the awaitTermination wait?

**Comment by harishreedharan** [comment]
In most cases, less than 5-10 seconds - just for the last in-flight batch to be done. This is not going to slow our unit test runs down. I mostly hit it in our internal end-to-end tests

File: external/flume/src/main/scala/org/apache/spark/streaming/flume/FlumePollingInputDStream.scala [hunk 1]
```diff
@@ -93,7 +93,11 @@ private[streaming] class FlumePollingReceiver(
 
   override def onStop(): Unit = {
     logInfo("Shutting down Flume Polling Receiver")
-    receiverExecutor.shutdownNow()
+    receiverExecutor.shutdown()
+    // Wait for a minute for the threads to die
```
Comment (vanzin) [inline external/flume/src/main/scala/org/apache/spark/streaming/flume/FlumePollingInputDStream.scala]: nit: "Wait up to" a minute.

**Comment by vanzin** [comment]
LGTM. I'd rephrase the description to say "up to a minute" since I was a little scared when reading the original one. "OMG all my apps are going to wait a minute before exiting!"

**Comment by tdas** [comment]
LGTM. Merging it master and branch 1.5

**Comment by tdas** [comment]
Actually, there were small conflicts in branch 1.5, so I only merged to master. Seems fine, not a highpriority bug that 1.5 should get it.
